index.js

```diff
@@ -1,34 +1,100 @@
 'use strict'
 
-var events = []
-var eventsFired = {}
 const EventEmitter = require('events')
-var instance = undefined
 
+const DEFAULT_NAMESPACE = 'app'
+const DEFAULT_TIMEOUT = 2000
+
+//object containing all namespaces
+let NS = {}
+
+/**
+ * Main class to handle all your dependencies
+ * with support for multiple modules/namespaces.
+ */
 class HeadRush extends EventEmitter {
-    constructor(eventsProvided) {
+    /**
+     * options object -> {
+     *  deps (required): array - list of strings specifying dependencies
+     *  namespace (optional): string - namespace/module to tie the dependencies to,
+     *  timeout (optional): integer - timeout for deps to be resolved within specified time
+     * }
+     */
+    constructor(options) {
         super()
-        if (!events.length) {
-            events = eventsProvided
+
+        if (!options) {
+            options = {}
         }
-        if (!events && !eventsProvided) {
-            throw new Error('Initial events need to be present to stun')
+
+        var currentNS = options.namespace
+        var timeout = options.timeout
+
+        if (!currentNS) {
```

Comment:
short circuit ?

Reply:
We'll run into issues with short-circuit if someone chooses to use int values for namespaces like 0,1, and so on.

If the value passed in options = 0, the default values will be assigned always.